pinot-spi/src/main/java/org/apache/pinot/spi/data/FieldSpec.java

```diff
@@ -563,4 +592,30 @@ public boolean isBackwardCompatibleWith(FieldSpec oldFieldSpec) {
             && EqualityUtils.isEqual(_dataType, oldFieldSpec._dataType)
             && EqualityUtils.isEqual(_isSingleValueField, oldFieldSpec._isSingleValueField);
   }
+
+  public static class FieldSpecMetadata {
+    public Map<FieldType, FieldTypeMetadata> _fieldTypes = new HashMap<>();
+
+    void put(FieldType type, FieldTypeMetadata metadata) {
+      _fieldTypes.put(type, metadata);
+    }
+  }
+
+  public static class FieldTypeMetadata {
+    public List<DataTypeMetadata> _allowedDataTypes = new ArrayList<>();
+
+    void add(DataTypeMetadata metadata) {
+      _allowedDataTypes.add(metadata);
+    }
+  }
+
+  public static class DataTypeMetadata {
+    public FieldSpec.DataType _name;
```

Comment:
DataType has a bunch of other useful properties that can be part of the response. 

```

    private final DataType _storedType;
    private final int _size;
    private final boolean _sortable;
    private final boolean _numeric;


Think we should include all that in the API result.

Reply:
Updated the wrapper structure to accommodate the data type properties.